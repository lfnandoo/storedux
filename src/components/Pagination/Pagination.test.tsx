import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react";

import { Pagination } from "./index";

describe("Pagination", () => {
  const onChangePage = jest.fn();
  const amount = 2;
  const page = 0;

  it("should render correctly", () => {
    const tree = renderer
      .create(
        <Pagination page={page} amount={amount} onChangePage={onChangePage} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should get current page", () => {
    const { getByTestId } = render(
      <Pagination page={page} amount={amount} onChangePage={onChangePage} />
    );

    const currentPage = page + 1;

    expect(getByTestId("current-page")).toHaveTextContent(
      currentPage.toString()
    );
  });

  it("should trigger onChangePage on prev", () => {
    const { getByTestId } = render(
      <Pagination page={1} amount={amount} onChangePage={onChangePage} />
    );

    fireEvent.click(getByTestId("prev-btn"));

    expect(onChangePage).toHaveBeenCalled();
  });

  it("should trigger onChangePage on next", () => {
    const { getByTestId } = render(
      <Pagination page={page} amount={amount} onChangePage={onChangePage} />
    );

    fireEvent.click(getByTestId("next-btn"));

    expect(onChangePage).toHaveBeenCalled();
  });

  it("should not trigger onChangePage on prev", () => {
    const { getByTestId } = render(
      <Pagination page={page} amount={amount} onChangePage={onChangePage} />
    );

    fireEvent.click(getByTestId("prev-btn"));

    expect(onChangePage).toHaveBeenCalledTimes(0);
  });

  it("should not trigger onChangePage on next", () => {
    const { getByTestId } = render(
      <Pagination page={page} amount={1} onChangePage={onChangePage} />
    );

    fireEvent.click(getByTestId("next-btn"));

    expect(onChangePage).toHaveBeenCalledTimes(0);
  });
});
