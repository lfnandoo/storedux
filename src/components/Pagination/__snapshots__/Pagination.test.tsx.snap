// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render correctly 1`] = `
<div
  className="sc-gtsrHT ePUhsz"
>
  <button
    className="sc-bdnxRM OPJfk"
    data-testid="prev-btn"
    disabled={true}
    onClick={[Function]}
  >
    Anterior
  </button>
  <p
    data-testid="current-page"
  >
    1
  </p>
  <button
    className="sc-bdnxRM OPJfk"
    data-testid="next-btn"
    disabled={false}
    onClick={[Function]}
  >
    Pr√≥xima
  </button>
</div>
`;
